db/l2db/l2db.go

```diff
@@ -296,6 +291,70 @@ func (l2db *L2DB) addTxs(txs []common.PoolL2Tx, checkPoolIsFull bool) error {
 	return tracerr.Wrap(err)
 }
 
+// UpdateTxAPI Update PoolL2Tx regular transactions in the pool.
+func (l2db *L2DB) UpdateTxAPI(tx *common.PoolL2Tx) error {
+	cancel, err := l2db.apiConnCon.Acquire()
+	defer cancel()
+	if err != nil {
+		return tracerr.Wrap(err)
+	}
+	defer l2db.apiConnCon.Release()
+	return tracerr.Wrap(l2db.updateTx(*tx))
+}
+
+// Update PoolL2Tx transaction in the pool
+func (l2db *L2DB) updateTx(tx common.PoolL2Tx) error {
+	const queryUpdatePart = `UPDATE tx_pool SET `
+
+	var queryUpdate string
+	var queryVars []interface{}
+
+	if tx.ToIdx != 0 {
+		queryUpdate += "to_idx = ?, "
+		queryVars = append(queryVars, tx.ToIdx)
+	}
+
+	if tx.ToEthAddr != common.EmptyAddr {
+		queryUpdate += "to_eth_addr = ?, "
+		queryVars = append(queryVars, tx.ToEthAddr)
+	}
+
+	if tx.ToBJJ != common.EmptyBJJComp {
+		queryUpdate += "to_bjj = ?, "
+		queryVars = append(queryVars, tx.ToBJJ)
+	}
+
+	if tx.MaxNumBatch != 0 {
```

Comment:
Updating to "empty" values is legit, so I'd suggest to just set the values the user is sending, just taking in consideration the existing behaviour for inserts, for instance, if I remember well, an empty Ethereum address (0x000...00) is stored as NULL in the DB